### PR TITLE
Copy paste issue fixed. Defect ID: 26002

### PR DESCRIPTION
**Pull request by GokulBothe**
Thank you for your contribution.
Before submitting this PR, please make sure:
- [x] PR description and commit message should describe the changes done in this PR
- [x] Verify the PR is pointing to correct branch i.e. Release or Beta branch if the code fix is for specific release , else point it to master
- [x] Latest Code from master or specific release branch is merged to your branch
- [x] No unwanted\commented\junk code is included
- [x] No new warning upon build solution
- [x] Code Summary\Comments are added to my code which explains what my code is doing
- [x] Existing unit test cases are passed
- [x] New Unit tests are added for your development
- [x] Sanity Tests are successfully executed for New and Existing Functionality
- [x] Verify that changes are compatible with all relevant browsers and platforms.
- [x] After creating pull request there should not be any conflicts
- [x] Resolve codacy comments
- [x] Builds and checks are passed before PR is sent for review
- [x] Resolve code review comments
- [x] Help Libray document is updated accordingly for Feature changes.

### CODE REVIEW

File: Ginger/GingerCore/GeneralLib/General.cs [hunk 1]
```diff
@@ -399,6 +400,15 @@ public static bool IsNameAlreadyexists(RepositoryItemBase repositoryItem, string
 
             switch (repositoryItem.GetItemType())
             {
+                case "HTMLReportConfiguration":
+                    if (WorkSpace.Instance.SolutionRepository.GetAllRepositoryItems<HTMLReportConfiguration>().Any(x => x.Name == resultValue))
+                    {
+                        Reporter.ToUser(eUserMsgKey.StaticWarnMessage, "Report Template with same name: " + "'" + resultValue + "'" + " already exists.");
```
Comment (Maheshkale447) [inline Ginger/GingerCore/GeneralLib/General.cs:406]: Use string interpolation to concat string

File: Ginger/GingerCore/GeneralLib/General.cs [hunk 2]
```diff
@@ -431,20 +441,37 @@ public static bool IsNameAlreadyexists(RepositoryItemBase repositoryItem, string
                     }
 
                     break;
-                case "EnvApplication":
-                    if (WorkSpace.Instance.SolutionRepository.GetAllRepositoryItems<ProjEnvironment>().Any(x => x.Applications.Any(y=>y.Name== resultValue)))
+                case "Environment":
+                    if (WorkSpace.Instance.SolutionRepository.GetAllRepositoryItems<ProjEnvironment>().Any(x => x.Name == resultValue))
                     {
-                        Reporter.ToUser(eUserMsgKey.StaticWarnMessage, "Application with same name: " + "'" + resultValue + "'" + " already exists.");
+                        Reporter.ToUser(eUserMsgKey.StaticWarnMessage, "Environment with same name: " + "'" + resultValue + "'" + " already exists.");
```
Comment (Maheshkale447) [inline Ginger/GingerCore/GeneralLib/General.cs:447]: Use string interpolation to concat string

File: Ginger/GingerCore/GeneralLib/General.cs [hunk 3]
```diff
@@ -431,20 +441,37 @@ public static bool IsNameAlreadyexists(RepositoryItemBase repositoryItem, string
                     }
 
                     break;
-                case "EnvApplication":
-                    if (WorkSpace.Instance.SolutionRepository.GetAllRepositoryItems<ProjEnvironment>().Any(x => x.Applications.Any(y=>y.Name== resultValue)))
+                case "Environment":
+                    if (WorkSpace.Instance.SolutionRepository.GetAllRepositoryItems<ProjEnvironment>().Any(x => x.Name == resultValue))
```
Comment (Maheshkale447) [inline Ginger/GingerCore/GeneralLib/General.cs:445]: Use string.equals instead of '=='

File: Ginger/GingerCore/GeneralLib/General.cs [hunk 4]
```diff
@@ -399,6 +400,15 @@ public static bool IsNameAlreadyexists(RepositoryItemBase repositoryItem, string
 
             switch (repositoryItem.GetItemType())
             {
+                case "HTMLReportConfiguration":
+                    if (WorkSpace.Instance.SolutionRepository.GetAllRepositoryItems<HTMLReportConfiguration>().Any(x => x.Name == resultValue))
```
Comment (Maheshkale447) [inline Ginger/GingerCore/GeneralLib/General.cs:404]: User string.Equals instead of ==

**Review by Maheshkale447 — CHANGES_REQUESTED** [review]

**Comment by GokulBothe** [comment]
As per suggestion changes have been made and pushed changes to git.

File: Ginger/GingerCore/GeneralLib/General.cs (hunk 1 above)
Comment (GokulBothe) [reply Ginger/GingerCore/GeneralLib/General.cs]: Done

File: Ginger/GingerCore/GeneralLib/General.cs (hunk 2 above)
Comment (GokulBothe) [reply Ginger/GingerCore/GeneralLib/General.cs]: Done

File: Ginger/GingerCore/GeneralLib/General.cs (hunk 3 above)
Comment (GokulBothe) [reply Ginger/GingerCore/GeneralLib/General.cs]: Done

File: Ginger/GingerCore/GeneralLib/General.cs (hunk 4 above)
Comment (GokulBothe) [reply Ginger/GingerCore/GeneralLib/General.cs]: Done

**Pull request merged**